pkg/controller/sesv2/configurationset/setup.go

```diff
@@ -0,0 +1,288 @@
+	if !isUpToDateDeliveryOptions(cr, resp) {
+		return false, nil
+	}
+
+	if !isUpToDateReputationOptions(cr, resp) {
+		return false, nil
+	}
+
+	if !isUpToDateSendingOptions(cr, resp) {
+		return false, nil
+	}
+
+	if !isUpToDateSuppressionOptions(cr, resp) {
+		return false, nil
+	}
+
+	if !isUpToDateTrackingOptions(cr, resp) {
+		return false, nil
+	}
+
+	return svcutils.AreTagsUpToDate(cr.Spec.ForProvider.Tags, resp.Tags)
+}
+
+// isUpToDateDeliveryOptions checks if DeliveryOptions Object are up to date
+func isUpToDateDeliveryOptions(cr *svcapitypes.ConfigurationSet, resp *svcsdk.GetConfigurationSetOutput) bool {
+	if cr.Spec.ForProvider.DeliveryOptions != nil && resp.DeliveryOptions != nil {
+		if awsclient.StringValue(cr.Spec.ForProvider.DeliveryOptions.SendingPoolName) != awsclient.StringValue(resp.DeliveryOptions.SendingPoolName) {
+			return false
+		}
+		if awsclient.StringValue(cr.Spec.ForProvider.DeliveryOptions.TLSPolicy) != awsclient.StringValue(resp.DeliveryOptions.TlsPolicy) {
+			return false
+		}
+	}
+	return true
+}
+
+// isUpToDateReputationOptions checks if ReputationOptions Object are up to date
+func isUpToDateReputationOptions(cr *svcapitypes.ConfigurationSet, resp *svcsdk.GetConfigurationSetOutput) bool {
+	if cr.Spec.ForProvider.ReputationOptions != nil && resp.ReputationOptions != nil {
+		if awsclient.BoolValue(cr.Spec.ForProvider.ReputationOptions.ReputationMetricsEnabled) != awsclient.BoolValue(resp.ReputationOptions.ReputationMetricsEnabled) {
+			return false
+		}
+	}
+	return true
+}
+
+// isUpToDateTrackingOptions checks if TrackingOptions Object are up to date
+func isUpToDateTrackingOptions(cr *svcapitypes.ConfigurationSet, resp *svcsdk.GetConfigurationSetOutput) bool {
+	if cr.Spec.ForProvider.TrackingOptions != nil && resp.TrackingOptions != nil {
+		if awsclient.StringValue(cr.Spec.ForProvider.TrackingOptions.CustomRedirectDomain) != awsclient.StringValue(resp.TrackingOptions.CustomRedirectDomain) {
+			return false
+		}
+	}
+	return true
+}
+
+// isUpToDateSuppressionOptions checks if SuppressionOptions Object are up to date
+func isUpToDateSuppressionOptions(cr *svcapitypes.ConfigurationSet, resp *svcsdk.GetConfigurationSetOutput) bool {
+	crSuppressedReasons := make([]*string, 0)
+	awsSuppressedReasons := make([]*string, 0)
+
+	if cr.Spec.ForProvider.SuppressionOptions != nil && cr.Spec.ForProvider.SuppressionOptions.SuppressedReasons != nil {
+		crSuppressedReasons = append(crSuppressedReasons, cr.Spec.ForProvider.SuppressionOptions.SuppressedReasons...)
+	}
+
+	if resp.SuppressionOptions != nil && resp.SuppressionOptions.SuppressedReasons != nil {
+		awsSuppressedReasons = resp.SuppressionOptions.SuppressedReasons
+	}
+
+	if len(crSuppressedReasons) != len(awsSuppressedReasons) {
+		return false
+	}
+
+	sortCmp := cmpopts.SortSlices(func(i, j *string) bool {
+		return aws.StringValue(i) < aws.StringValue(j)
+	})
+
+	return cmp.Equal(crSuppressedReasons, awsSuppressedReasons, sortCmp)
```

Comment:
I think a simple for loop checking for `StringValue` would be less costly here than reflection.

Reply:
for loop method will be more verbose (less clarity) and input from custom resource may not be in order